### ISSUE TITLE
vertex-object-renderers performance regression on macOS/arm64

### ISSUE THREAD

**Issue by kintel** [openscad/openscad#4504]
Current master in Release mode on an 1 year old MacBook Pro arm64 (Apple M1 Pro):
```
Without `--enable=vertex-object-renderers`:
ctest -j10  341.64s user 136.32s system 577% cpu 1:22.79 total

With `--enable=vertex-object-renderers`:
ctest -j10  552.33s user 410.91s system 338% cpu 4:44.35 total
```
Enabling various other VBO-related feature may change these numbers, but opening this ticket to track this.

Re-running this in October 2023 yields even worse results:
```
time ctest -j10

# Default (--enable=vertex-object-renderers)
ctest -j10  2309.63s user 1310.52s system 381% cpu 15:50.01 total


# No VBO
ctest -j10  2067.78s user 1502.90s system 348% cpu 17:03.15 total
```

A comparative Linux machine runs tests in ~90 seconds.

**Comment by kintel** [comment]
The main culprit seems to be that `glBufferSubData()` is being called very often (typically every 12 bytes), and this call is comparatively expensive on macOS.

Suggested fix: Upload buffers using a single (or very few) `glBufferSubData()` or `glBufferData()` calls.

**Comment by kintel** [comment]
Fixed by #4787
